invoices/invoiceregistry_test.go

```diff
@@ -0,0 +1,150 @@
+package invoices
+
+import (
+	"crypto/sha256"
+	"github.com/btcsuite/btcd/chaincfg"
+	"github.com/lightningnetwork/lnd/channeldb"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"io/ioutil"
+	"testing"
+	"time"
+
+	"github.com/fortytw2/leaktest"
```

Comment:
is this import meant to be permanent? looks like the lib isn't compatible with `t.Parallel()`, so although it may work here it would be hard to use it elsewhere in the codebase

Reply:
I was planning to leave it in. Is it a problem if we just don't use `t.Parallel()`? I try to make all my tests without any `time.Sleep`, so they are fast.

Reply:
I think the use of parallelism greatly speeds up the unit tests, especially on travis and/or with the race detector running. attempting to not use sleeps is definitely the way to go, though there are certain areas of the codebase that test timing assumptions and it probably can't be avoided. IMO this would be a good tool to have when debugging, but don't know if it needs to be committed in mainline. perhaps @Roasbeef has some thoughts?

Reply:
Interesting. Will this detect cases of for instance a client goroutine within the invoiceregistry not exiting before the test is over?

Reply:
yes

Reply:
> Is it a problem if we just don't use t.Parallel()? 

That'd make the _entire_ test suite run slower, not to mention the additional impact on tests running w/ the race condition detector. We can use external tooling to catch stray goroutine tools, would rather not further extend the execution time of our integration tests.

Reply:
See comment above